src/TodoApp.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {
+  FormEvent,
+  useEffect,
+  useState,
+} from 'react';
+import { useLocation } from 'react-router-dom';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { Footer } from './components/Footer';
+import { FilterTypes } from './types/FilterTypes';
+
+export const TodoApp: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(() => {
+    const localTodos = localStorage.getItem('todos');
+
+    try {
+      return localTodos ? JSON.parse(localTodos) : [];
+    } catch {
+      return [];
+    }
```

Comment:
For what this catch? In 17 line you check `localTodos` and if it undefined you return `[]`